ekncontent/ekncontent/eknc-domain.c

```diff
@@ -830,8 +858,13 @@ on_xapian_query_response (GObject *source,
           return;
         }
 
+      /* Note that we pass another closure here instead of just the task
```

Comment:
You don't have to encapsulate the task in a closure data struct - use `g_task_set_task_data` instead.